src/rebar3-nix-bootstrap

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env escript
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil -*-
+%%! -smp enable
+%%% ---------------------------------------------------------------------------
+%%% @doc
+%%% The purpose of this command is to prepare a rebar3 project so that
+%%% rebar3 understands that the dependencies are all already
+%%% installed. If you want a hygienic build on nix then you must run
+%%% this command before running rebar3. I suggest that you add a
+%%% `Makefile` to your project and have the bootstrap command be a
+%%% dependency of the build commands. See the nix documentation for
+%%% more information.
+%%%
+%%% This command designed to have as few dependencies as possible so
+%%% that it can be a dependency of root level packages like rebar3. To
+%%% that end it does many things in a fairly simplistic way. That is
+%%% by design.
+%%%
+%%% ### Assumptions
+%%%
+%%% This command makes the following assumptions:
+%%%
+%%% * It is run in a nix-shell or nix-build environment
+%%% * that all dependencies have been added to the ERL_LIBS
+%%%   Environment Variable
+
+-record(data, {version
+              , registry_only = false
+              , compile_ports = false
+              , erl_libs
+              , root
+              , name
+              , registry_snapshot}).
+
+-define(HEX_REGISTRY_PATH, ".cache/rebar3/hex/default/registry").
+
+main(Args) ->
+    {ok, ArgData} = parse_args(Args),
+    {ok, RequiredData} = gather_required_data_from_the_environment(ArgData),
+    do(RequiredData).
+
+%% @doc
+%% This actually runs the command. There are two modes 'register_only'
+%% where the register is created from hex and everything else.
+-spec do(#data{}) -> ok.
+do(RequiredData = #data{registry_only = true}) ->
+    ok = bootstrap_registry(RequiredData);
+do(RequiredData) ->
+    ok = bootstrap_registry(RequiredData),
+    ok = bootstrap_configs(RequiredData),
+    ok = bootstrap_libs(RequiredData).
+
+%% @doc
+%% Argument parsing is super simple only because we want to keep the
+%% dependencies minimal. For now there can be two entries on the
+%% command line, "register-only" and "compile-ports"
+-spec parse_args([string()]) -> #data{}.
+parse_args(Args) ->
+    {ok, #data{registry_only = lists:member("registry-only", Args)
+              , compile_ports = lists:member("compile-ports", Args)}}.
```

Comment:
This does not seem to be mentioned at all from https://github.com/NixOS/nixpkgs/compare/master...ericbmerritt:erlang-improvements

Shouldn't it be called from there? Or maybe replace arg-checking with env-checking?